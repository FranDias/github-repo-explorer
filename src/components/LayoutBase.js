import React from "react";
import PropTypes from "prop-types";

const LayoutBase = ({ header, children }) => {
  return (
    <div className="layout-base">
      {header && <header>{header}</header>}
      {children && <div className="body">{children}</div>}
    </div>
  );
};

LayoutBase.propTypes = {
  header: PropTypes.node,
  children: PropTypes.node
};

export default LayoutBase;
