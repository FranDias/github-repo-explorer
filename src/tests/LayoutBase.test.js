import React from "react";
import renderer from "react-test-renderer";
import LayoutBase from "../components/LayoutBase";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("LayoutBase", () => {
  function HeaderContent() {
    return <div />;
  }
  HeaderContent.displayName = "HeaderContent";

  function BodyContent() {
    return <body />;
  }
  BodyContent.displayName = "BodyContent";

  it("matches its snapshot with minimal props", () => {
    const component = renderer.create(<LayoutBase />);
    expect(component).toMatchSnapshot();
  });

  it("matches its snapshot with header props", () => {
    const component = renderer.create(<LayoutBase header={HeaderContent} />);
    expect(component).toMatchSnapshot();
  });

  it("matches its snapshot with children props", () => {
    const component = renderer.create(<LayoutBase>{BodyContent}</LayoutBase>);
    expect(component).toMatchSnapshot();
  });

  describe("renders", () => {
    let component;
    beforeEach(() => {
      component = shallow(
        <LayoutBase header={HeaderContent}>{BodyContent}</LayoutBase>
      );
    });
    debugger
    it("a header", () => {
      expect(component.find(HeaderContent).exists()).toBe(true);
    });

    it("body", () => {
      expect(component.find(BodyContent).exists()).toBe(true);
    });
  });
});
