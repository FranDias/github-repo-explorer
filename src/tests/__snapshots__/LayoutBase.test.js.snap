// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutBase matches its snapshot with children props 1`] = `
<div
  className="layout-base"
>
  <div
    className="body"
  >
    <body />
  </div>
</div>
`;

exports[`LayoutBase matches its snapshot with header props 1`] = `
<div
  className="layout-base"
>
  <header>
    <div />
  </header>
</div>
`;

exports[`LayoutBase matches its snapshot with minimal props 1`] = `
<div
  className="layout-base"
/>
`;
